### PR TITLE
btrfs-progs: add udevSupport arg

### PR DESCRIPTION
**Pull request by goertzenator**
###### Description of changes
Add argument `udevSupport` (edit: was `disableUdev`) to give the option of disabling udev support.

I maintain a small embedded OS using Nix and recent changes in `btrfs-progs` caused it to bring in `systemd`.  This OS doesn't use `udev` at all, let alone `systemd`, so this PR allows `udev` support to be disabled entirely.

###### Things done
I have verified that `btrfs-progs` no longer brings in `systemd` when `udevSupport = false`.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/btrfs-progs/default.nix [hunk 1]
```diff
@@ -31,7 +33,11 @@ stdenv.mkDerivation rec {
     install -v -m 444 -D btrfs-completion $out/share/bash-completion/completions/btrfs
   '';
 
-  configureFlags = lib.optional stdenv.hostPlatform.isMusl "--disable-backtrace --disable-libudev";
+  configureFlags = let
+      musl = stdenv.hostPlatform.isMusl;
+    in
+      (lib.optional musl "--disable-backtrace") ++
+      (lib.optional (musl || disableUdev) "--disable-libudev");
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/btrfs-progs/default.nix:40]:
```suggestion
  configureFlags = lib.optional musl "--disable-backtrace"
   ++ lib.optional (stdenv.isMusl || disableUdev) "--disable-libudev";
```

File: pkgs/tools/filesystems/btrfs-progs/default.nix [hunk 2]
```diff
@@ -4,6 +4,8 @@
 , acl, attr, e2fsprogs, libuuid, lzo, udev, zlib
 , runCommand, btrfs-progs
 , gitUpdater
+
+, disableUdev ? false
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/btrfs-progs/default.nix:8]:
```suggestion
, udevSupport ? true
```

File: pkgs/tools/filesystems/btrfs-progs/default.nix (hunk 1 above)
Comment (goertzenator) [reply pkgs/tools/filesystems/btrfs-progs/default.nix]: I've changed the structure as suggested but just to confirm, this should stay as `stdenv.hostPlatform.isMusl`, right?

Comment (SuperSandro2000) [reply pkgs/tools/filesystems/btrfs-progs/default.nix]: On is an alias to the other.

**Comment by SuperSandro2000** [comment]
Please follow the contributing guide when naming your comments and rebase to fix the merge conflict.

**Comment by goertzenator** [comment]
This should be ready to merge.  Please advise if there's anything more I need to do to get it merge-ready.

File: pkgs/tools/filesystems/btrfs-progs/default.nix [hunk 3]
```diff
@@ -4,6 +4,8 @@
 , acl, attr, e2fsprogs, libuuid, lzo, udev, zlib
 , runCommand, btrfs-progs
 , gitUpdater
+
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/btrfs-progs/default.nix:7]:
```suggestion
```

**Pull request merged**